components/Filters/spamlist/MoveEmailFilteredList.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { SmallButton, Icon, useApiWithoutResult, useNotifications } from 'react-components';
+import { updateIncomingDefault } from 'proton-shared/lib/api/incomingDefaults';
+import { noop } from 'proton-shared/lib/helpers/function';
+import { MAILBOX_IDENTIFIERS } from 'proton-shared/lib/constants';
+
+const BLACKLIST_TYPE = +MAILBOX_IDENTIFIERS.spam;
+const WHITELIST_TYPE = +MAILBOX_IDENTIFIERS.inbox;
+
+function MoveEmailFilteredList({ type, dest, email, className, onClick }) {
+    const { createNotification } = useNotifications();
+    const { request } = useApiWithoutResult(updateIncomingDefault);
+    const iconName = type === 'whitelist' ? `arrow-right` : `arrow-left`;
+
+    const I18N = {
+        blacklist: c('Title').t`blacklist`,
+        whitelist: c('Title').t`whitelist`
+    };
+
+    const handleClick = async () => {
+        const { Email, ID } = email;
+        const Location = dest === 'whitelist' ? WHITELIST_TYPE : BLACKLIST_TYPE;
+        const { IncomingDefault: data } = await request(ID, { Location });
```

Comment:
I think the logic of this function would be better handled in `SpamFiltersContainer` because it would allow to set the temporary list before starting the request / and reset it if it fails etc.

and it could return a promise and to disable this button while the action is running

Reply:
pagination comes from the API so we can't use do a temp list as 95% of the time for the user it's the same as seeing a delete action.